src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2309,42 +2347,35 @@ ProcessTargetListForWorkerQuery(List *targetEntryList,
 								QueryTargetList *queryTargetList,
 								QueryGroupClause *queryGroupClause)
 {
-	ListCell *targetEntryCell = NULL;
 	WorkerAggregateWalkerContext workerAggContext = {
 		.extendedOpNodeProperties = extendedOpNodeProperties,
 	};
 
 	/* iterate over original target entries */
-	foreach(targetEntryCell, targetEntryList)
+	TargetEntry *originalTargetEntry = NULL;
+	foreach_ptr(originalTargetEntry, targetEntryList)
 	{
-		TargetEntry *originalTargetEntry = (TargetEntry *) lfirst(targetEntryCell);
 		Expr *originalExpression = originalTargetEntry->expr;
 		List *newExpressionList = NIL;
-		bool hasAggregates = contain_aggs_of_level((Node *) originalExpression, 0);
-		bool hasWindowFunction = contain_window_function((Node *) originalExpression);
 
 		/* reset walker context */
 		workerAggContext.expressionList = NIL;
 		workerAggContext.createGroupByClause = false;
 
 		/*
-		 * If the query has a window function, we currently assume it's safe to push
-		 * down the target list.
-		 *
 		 * If there are aggregates without a GROUP BY on the distribution column
 		 * then the results of those aggregates need to be combined on the coordinator.
 		 * In that case we rewrite the expressions using WorkerAggregateWalker.
```

Comment:
It feels this comment is not covering what we are testing anymore. The `CanPushDownExpression` has more checks than is described here.

Instead I think we can replace the comment with;

```suggestion
		 * If we can push down the expression we copy the expression to the targetlist of the worker query.
		 * Otherwise we find all column references (?) from the expression, add these to the targetlist so the
		 * expression can be computed on the coordinator instead of the worker.
```